lesson11/main.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const body = document.getElementById("body");
+const wrapper = document.createElement("div");
+body.appendChild(wrapper);
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+const endLoading = () => wrapper.remove();
+const jsonURL = "https://jsondata.okiba.me/v1/json/KKl45211103113313";
+
+async function fetchData() {
+   try {
+      const response = await fetch(jsonURL);
+      const json = await response.json();
+      return json.data;
+   } catch(error) {
+      throw new Error(`データの取得に失敗しました：${error}`);
+   } finally {
+      console.log('処理が終了しました。');
+   }
+}
+
+async function init() {
+   startLoading();
+   try {
+      const result = await fetchData();
+      createListElements(result);
+   } catch(e) {
+      console.log('データが取得できませんでした');
+      console.error(e);
```

Comment:
nits
I think it's better to display the error in the browser to let the user know when it happens.
console.log may be for developers, but users will be baffled to see a blank screen.

Reply:
Thank you for your advise.
It's true that if a blank screen appears, it's hard to understand what it means.
In that case, I changed to appear the text.
7519e25